docs/reference/snapshot-restore/index.asciidoc

```diff
@@ -57,17 +57,21 @@ when migrating between versions.
 
 A snapshot contains a copy of the on-disk data structures that make up an
 index or a data stream's backing indices. This means that snapshots can only be restored to versions of
-{es} that can read the indices:
-
-* A snapshot of an index created in 6.x can be restored to 7.x.
-* A snapshot of an index created in 5.x can be restored to 6.x.
-* A snapshot of an index created in 2.x can be restored to 5.x.
-* A snapshot of an index created in 1.x can be restored to 2.x.
-
-Conversely, snapshots of indices created in 1.x **cannot** be restored to 5.x
-or 6.x, snapshots of indices created in 2.x **cannot** be restored to 6.x
-or 7.x, and snapshots of indices created in 5.x **cannot** be restored to 7.x
-or 8.x.
+{es} that can read the indices.
+
+The following table indicates snapshot compatibility between versions.
```

Comment:
This is a confusing table to me to be honest: 

For starters:
1. I'm not sure we actually should enumerate all the possible version configurations here.
2. Why would `5.x` -> `5.x` not be a "yes"?

Maybe it would be more helpful to make the two constraints we have crystal clear and shorten this section. In the end these two things are the rules (see my comment below also):

1. You cannot restore a snapshot taken by a version `x` into a cluster of version `y` if x is newer than `y`.
2. You cannot restore an index that was created with a version `x` into a cluster of version `y` if `y` is more than one major version newer than `x`.

That's all. I think the examples might have made sense in the past because we jumped from `2.x` to `5.x` but are probably unnecessary now that `7.x` only has to concern itself with `6.x` and newer anyway?

Reply:
@original-brownbear, thanks for your feedback 👍 I agree that we should add compatibility between the same versions, and also explain the difference between restoring snapshots and indices. However, I think that the table provides a clear visual indicator that allows users to quickly scan for the compatibility.

How does this version look to you?
![image](https://user-images.githubusercontent.com/25848033/88593175-2a4cbf80-d02d-11ea-802a-28b81fbe9587.png)